### PR TITLE
feat: available seasonal plants

### PR DESCRIPTION
**Pull request by Bushuo**
Ref: #453 

## Basics

- [X] Details of what you changed are in commit messages.
- [ ] References to issues, e.g. `close #X`, are in the commit messages.
- [ ] The buildserver is happy.
- [X] The PR is rebased with current master.

## Checklist

- [ ] I fully described what my PR does in the documentation
      (not in the PR description)
- [ ] I fixed all affected documentation
- [ ] I fixed all affected decisions
- [ ] I added unit tests for my code
- [ ] I added code comments, logging, and assertions as appropriate
- [ ] I translated all strings visible to the user
- [ ] I mentioned [every code or binary](/.reuse/dep5) not directly written or done by me in [reuse syntax](https://reuse.software/)
- [ ] I created left-over issues for things that are still to be done
- [ ] Code is conforming to [our Architecture](/doc/architecture)
- [ ] Code is conforming to [our Guidelines](/doc/guidelines)
      (exceptions are documented)
- [ ] Code is consistent to [our Design Decisions](/doc/decisions)

## Review

- [ ] I've tested the code
- [ ] I've read through the whole code
- [ ] Examples are well chosen and understandable

### CODE REVIEW

File: backend/src/model/dto.rs [hunk 1]
```diff
@@ -34,24 +34,41 @@ pub struct ConfigDto {
...
+#[allow(clippy::missing_docs_in_private_items)]
 #[typeshare]
 #[derive(Serialize, Deserialize, ToSchema)]
 pub struct SeedDto {
+    /// The record id of the seed.
     pub id: i32,
+    /// An additional name for the seed.
     pub name: String,
+    /// The variety of the seed. Currently unused.
     pub variety: Option<String>,
```
Comment (markus2330) [inline backend/src/model/dto.rs:47]: Leftover, I created #462

```suggestion
    /// The variety of the seed. Should be removed, see #462.
    pub variety: Option<String>,
```

**Review by markus2330 — APPROVED** [review]
Great job! Comments can also be addressed later, please create follow-up issue if they cannot be fixed easily. I wasn't able to test it, as the feature seems to be not yet available in frontend and I also couldn't find it in Swagger UI.

**Comment by markus2330** [comment]
jenkins build please

**Comment by markus2330** [comment]
This should be the last new feature for our goal, now let us concentrate on filling gaps, integrating and fixing bugs!

File: backend/src/model/dto.rs [hunk 2]
```diff
@@ -34,24 +34,41 @@ pub struct ConfigDto {
     pub version: i32,
 }
 
-#[allow(clippy::missing_docs_in_private_items)] // TODO: See #97.
+/// Represents seeds of a user.
+#[allow(clippy::missing_docs_in_private_items)]
```
Comment (kitzbergerg) [inline backend/src/model/dto.rs:38]: I think you can now remove this line as it is documented.

File: backend/src/controller/planting_suggestions.rs [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+    service,
+};
+
+/// Endpoint for listing suggestions.
+/// Suggestions are `Plants` that are suitable for a given `Map`.
+///
+/// # Errors
+/// * If the connection to the database could not be established.
+#[utoipa::path(
+    context_path = "/api/maps//{map_id}/layers/plant/suggestionss",
```
Comment (kitzbergerg) [inline backend/src/controller/planting_suggestions.rs]: typo

File: backend/src/model/entity.rs [hunk 4]
```diff
@@ -651,24 +651,40 @@ pub struct Plants {
     pub edible_uses_de: Option<String>,
 }
 /// The `Seed` entity.
-#[allow(clippy::missing_docs_in_private_items)] // TODO: See #97.
+#[allow(clippy::missing_docs_in_private_items)]
```
Comment (kitzbergerg) [inline backend/src/model/entity.rs:654]: Can be removed

**Comment by Bushuo** [comment]
@markus2330 can we stop the CI jobs except the last one please 😢 Sorry, I pushed too early

**Comment by markus2330** [comment]
NP, I killed all but https://build.libelektra.org/job/PermaPlanT-Folder/job/PermaplanT/job/PR-456/12/

Btw. fel115 and lukashartl should also be able to do that :wink:

**Comment by markus2330** [comment]
Now we have conflicts here.

**Comment by Bushuo** [comment]
> Now we have conflicts here.

@markus2330 
I fixed the conflicts. Can we merge this?

**Pull request merged**